src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/OffenderSearchRestClient.java

```diff
@@ -75,34 +73,25 @@ public OffenderSearchRestClient(@Qualifier("offenderSearchWebClient") WebClient
         }
     }
 
-    private boolean validate(String fullName, LocalDate dateOfBirth) {
-        if(dateOfBirth == null) {
+    private boolean validate(Name name, LocalDate dateOfBirth) {
+        if (dateOfBirth == null) {
             log.error(ERROR_NO_DATE_OF_BIRTH);
-            eventBus.post(OffenderSearchValidationFailureEvent.builder()
-                    .failureMessage(ERROR_NO_DATE_OF_BIRTH)
-                    .fullName(fullName)
-                    .build());
             return false;
         }
 
-        if(StringUtil.isNullOrEmpty(fullName)) {
+        if (name == null || StringUtils.isEmpty(name.getSurname())) {
             log.error(ERROR_NO_NAME);
-            eventBus.post(OffenderSearchValidationFailureEvent.builder()
-                    .failureMessage(ERROR_NO_NAME)
-                    .fullName(fullName)
-                    .dateOfBirth(dateOfBirth)
-                    .build());
             return false;
         }
         return true;
     }
 
     private Mono<? extends SearchResponse> handleError(Throwable throwable, MatchRequest body) {
-        log.error(throwable.getMessage());
         String incomingMessage = null;
         try {
             // Would rather this didn't block but also want to provide a helpful error message
             // TODO: come up with a better solution
+            // We could send a toString representation of the messageBody or do the parsing to JSON in the EventBus
```

Comment:
Good idea, I'll raise a ticket to do this and review the error handling whilst we're at it. Only a tiny job but would rather not block progress on this ticket to do it.

Reply:
Created [PIC-828](https://dsdmoj.atlassian.net/browse/PIC-828) for this